kpi_dashboard/models/kpi_kpi.py

```diff
@@ -84,6 +87,30 @@ def write(self, vals):
             vals["value_last_update"] = fields.Datetime.now()
         return super().write(vals)
 
+    def _get_code_input_dict(self):
+        return {
+            "self": self,
+            "model": self.browse(),
+        }
+
+    def _compute_value_code(self):
+        new_cr = Registry(self.env.cr.dbname).cursor()
+        try:
```

Comment:
This tries to disallow something like self.unlink()

We are trying to find a better option...

Reply:
Put it inside a savepoint and do rollback after:
```python
self.env.cr.execute("savepoint kpi_formula_%" % self.id)
safe_eval(...)
self.env.cr.execute("rollback to kpi_formula_%" % self.id)
```

Reply:
Thanks @pedrobaeza your solution is good, however the savepoint is not perfect. I mean, if the user puts 
```
self.unlink()
self.env.cr.commit()
```
the unlink will be performed. However, the same happened with my solution and this is already cleaner, so I will keep it if nothing better can be found.

Reply:
Why don't you simply look for `.commit()` in a constraint and forbids that string?

Reply:
That will probably be the best combined with the savepoint